src/mygrad/tensor_base.py

```diff
@@ -25,9 +36,12 @@
 from mygrad.tensor_core_ops.indexing import GetItem, SetItem
 from mygrad.tensor_manip.array_shape.ops import Flatten
 from mygrad.tensor_manip.transpose_like.ops import Tensor_Transpose_Property
+from mygrad.typing import ArrayLike
 
 __all__ = ["Tensor"]
 
+T = TypeVar("T")
```

Comment:
I'd define all your typevars in `mygrad.typing`; in hindsight we should have made a `hypothesis._typing` module.